linux_os/guide/system/software/integrity/fips/system_boot_in_fips_mode/oval/shared.xml

```diff
@@ -0,0 +1,30 @@
+<def-group>
+    <definition class="compliance" id="{{{ rule_id }}}" version="1">
+        {{{ oval_metadata("The system must be booted with fips=1 and /proc/cmdline must not contain fips=0") }}}
+        <criteria operator="AND">
+            <criterion comment="FIPS mode is enabled" test_ref="test_{{{ rule_id }}}_mode_exists" />
+            <criterion comment="FIPS mode is not disabled" test_ref="test_{{{ rule_id }}}_not_disabled" />
+        </criteria>
+    </definition>
+
+    <ind:textfilecontent54_test check="all" check_existence="all_exist" comment="FIPS mode is enabled" id="test_{{{ rule_id }}}_mode_exists" version="1">
+        <ind:object object_ref="obj_{{{ rule_id }}}_mode_exists" />
+    </ind:textfilecontent54_test>
+
+    <ind:textfilecontent54_test check="all" check_existence="none_exist" id="test_{{{ rule_id }}}_not_disabled" version="1" comment="FIPS mode isn't disable">
+        <ind:object object_ref="obj_{{{ rule_id }}}_not_disabled" />
+    </ind:textfilecontent54_test>
+
+    <ind:textfilecontent54_object id="obj_{{{ rule_id }}}_mode_exists" version="1">
+        <ind:filepath>/proc/cmdline</ind:filepath>
+        <ind:pattern operation="pattern match">.+fips*=1.+</ind:pattern>
```

Comment:
Why does the pattern have a `*` between `fips` and the equals sign?

If this is a regex (which the starting `.+` seems to indicate), then this would match `fipsssss=1` or `fip=1`, which don't enable FIPS mode.

Also, this would match `notreallyfips=1` due to the `.+` before; That regex should probably instead be anchored by matching a word boundary, or explicitly search for whitespace before and after `fips=1`.

However, there's really no difference between the kernel command line containing `fips=1` and `/proc/sys/crypto/fips_enabled` containing `1`, so I'd propose checking for the latter instead, because it's much easier to do.